contracts/PLCRFactory.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.4.20;
+
+import "tokens/eip20/EIP20.sol";
+import "./PLCRVoting.sol";
+import "./ProxyFactory.sol";
+
+contract PLCRFactory {
+
+  event newPLCR(address creator, EIP20 token, address plcr);
+
+  struct PLCR {
+    address creator;
+    EIP20 token;
+    PLCRVoting plcr;
+  }
+
+  ProxyFactory proxyFactory;
+  PLCRVoting canonizedPLCR;
+
+  /// @dev constructor deploys a new canonical PLCRVoting contract and a proxyFactory.
+  constructor() {
+    canonizedPLCR = new PLCRVoting();
+    proxyFactory = new ProxyFactory();
+  }
+
+  /*
+  @dev deploys and initializes a new PLCRVoting contract that consumes a token at an address
+  supplied by the user.
+  @param _token an EIP20 token to be consumed by the new PLCR contract
+  */
+  function newPLCRBYOToken(EIP20 _token) public returns (PLCRVoting) {
```

Comment:
could this be simplified?

```sol
function newPLCRBYOToken(EIP20 _token) public returns (PLCRVoting) {
    address creator = msg.sender;
    EIP20 token = _token;
    PLCRVoting plcr = PLCRVoting(proxyFactory.createProxy(canonizedPLCR, ""));

    plcr.init(token);
    emit newPLCR(creator, token, plcr);
    return plcr;
}
```

or is the current implementation more efficient?